check/src/main/java/ru/tinkoff/qa/neptune/check/InnerMatcher.java

```diff
@@ -0,0 +1,36 @@
+package ru.tinkoff.qa.neptune.check;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+
+import static java.lang.String.format;
+import static java.lang.System.lineSeparator;
+
+class InnerMatcher<T> extends BaseMatcher<T> {
+
+    private static final String LINE_SEPARATOR = lineSeparator();
+    private final String description;
+    private final Matcher<? super T> criteria;
+
+    InnerMatcher(String description, Matcher<? super T> criteria) {
```

Comment:
Perhaps you forget to write checkArgument() methods for parameter, cuz it's an entry point of them.